tests/adapters/switches/juniper_mx_test.py

```diff
@@ -2098,6 +2098,289 @@ def test_set_access_vlan_on_interface_that_already_has_it_does_nothing(self):
+                <name>PATATE</name>
+                <vlan-id>1000</vlan-id>
+              </domain>
+            </bridge-domains>
+            <interfaces>
+              <interface>
+                <name>xe-0/0/6</name>
+                <unit>
+                  <name>0</name>
+                  <family>
+                    <bridge>
+                    </bridge>
+                  </family>
+                </unit>
+              </interface>
+            </interfaces>
+        """))
+
+        self.netconf_mock.should_receive("edit_config").once().with_args(target="candidate", config=is_xml("""
+            <config>
+              <configuration>
+                <interfaces>
+                  <interface>
+                    <name>xe-0/0/6</name>
+                    <unit>
+                      <name>0</name>
+                      <family>
+                        <bridge>
+                          <interface-mode>access</interface-mode>
+                            <vlan-id>1000</vlan-id>
+                        </bridge>
+                      </family>
+                    </unit>
+                  </interface>
+                </interfaces>
+              </configuration>
+            </config>
+        """)).and_return(an_ok_response())
+
+        self.switch.set_access_vlan("xe-0/0/6", 1000)
+
+    def test_set_access_vlan_on_interface_replaces_vlan_id(self):
+        self.netconf_mock.should_receive("get_config").with_args(source="candidate", filter=is_xml("""
+            <filter>
+              <configuration>
+                <interfaces/>
+                <bridge-domains/>
+              </configuration>
+            </filter>
+        """)).and_return(a_configuration("""
+            <bridge-domains>
+              <domain>
+                <name>PATATE</name>
+                <vlan-id>1000</vlan-id>
+              </domain>
+              <domain>
+                <name>PATATE2</name>
+                <vlan-id>2000</vlan-id>
+              </domain>
+            </bridge-domains>
+            <interfaces>
+              <interface>
+                <name>xe-0/0/6</name>
+                <unit>
+                  <name>0</name>
+                  <family>
+                    <bridge>
+                      <interface-mode>access</interface-mode>
+                      <vlan-id>2000</vlan-id>
+                    </bridge>
+                  </family>
+                </unit>
+              </interface>
+            </interfaces>
+        """))
+
+        self.netconf_mock.should_receive("edit_config").once().with_args(target="candidate", config=is_xml("""
+            <config>
+              <configuration>
+                <interfaces>
+                  <interface>
+                    <name>xe-0/0/6</name>
+                    <unit>
+                      <name>0</name>
+                      <family>
+                        <bridge>
+                          <vlan-id operation="delete"/>
```

Comment:
you don't really need this statement, does work though...  but not if the operation=delete comes after the vlan-id, then it will remove... i don't really like that we depend on the order of the statements in an XML structure